src/Compilers/CSharp/Portable/Symbols/PublicModel/Symbol.cs

```diff
@@ -0,0 +1,245 @@
+            }
+
+            return builder.ToImmutableAndFree();
+        }
+
+        protected static ImmutableArray<TypeWithAnnotations> ConstructTypeArguments(ImmutableArray<ITypeSymbol> typeArguments, ImmutableArray<CodeAnalysis.NullableAnnotation> typeArgumentNullableAnnotations)
+        {
+            if (typeArguments.IsDefault)
+            {
+                throw new ArgumentException(nameof(typeArguments));
+            }
+
+            int n = typeArguments.Length;
+            if (!typeArgumentNullableAnnotations.IsDefault && typeArgumentNullableAnnotations.Length != n)
+            {
+                throw new ArgumentException(nameof(typeArgumentNullableAnnotations));
+            }
+
+            var builder = ArrayBuilder<TypeWithAnnotations>.GetInstance(n);
+            for (int i = 0; i < n; i++)
+            {
+                var type = typeArguments[i].EnsureCSharpSymbolOrNull(nameof(typeArguments));
+                var annotation = typeArgumentNullableAnnotations.IsDefault ? NullableAnnotation.Oblivious : typeArgumentNullableAnnotations[i].ToInternalAnnotation();
+                builder.Add(TypeWithAnnotations.Create(type, annotation));
+            }
+
+            return builder.ToImmutableAndFree();
+        }
+
+        ISymbol ISymbol.OriginalDefinition
+        {
+            get
+            {
+                return UnderlyingSymbol.OriginalDefinition.GetPublicSymbol();
+            }
+        }
+
+        ISymbol ISymbol.ContainingSymbol
+        {
+            get
+            {
+                return UnderlyingSymbol.ContainingSymbol.GetPublicSymbol();
+            }
+        }
+
+        INamedTypeSymbol ISymbol.ContainingType
+        {
+            get
+            {
+                return UnderlyingSymbol.ContainingType.GetPublicSymbol();
+            }
+        }
+
+        public sealed override int GetHashCode()
+        {
+            return UnderlyingSymbol.GetHashCode();
+        }
+
+        public sealed override bool Equals(object obj)
+        {
+            return this.Equals(obj as Symbol, CodeAnalysis.SymbolEqualityComparer.Default);
+        }
+
+        bool IEquatable<ISymbol>.Equals(ISymbol other)
+        {
+            return this.Equals(other as Symbol, CodeAnalysis.SymbolEqualityComparer.Default);
+        }
+
+        bool ISymbol.Equals(ISymbol other, CodeAnalysis.SymbolEqualityComparer equalityComparer)
+        {
+            return this.Equals(other as Symbol, equalityComparer);
+        }
+
+        protected bool Equals(Symbol other, CodeAnalysis.SymbolEqualityComparer equalityComparer)
+        {
+            return other is object && UnderlyingSymbol.Equals(other.UnderlyingSymbol, equalityComparer.CompareKind);
+        }
+
+        ImmutableArray<Location> ISymbol.Locations
+        {
+            get
+            {
+                return UnderlyingSymbol.Locations;
```

Comment:
Some of the properties in this file use the fully-spelled out syntax, and some use `=>`. We should use one style, I'd prefer `=>`.

Reply:
> Some of the properties in this file use the fully-spelled out syntax, and some use =>. We should use one style, I'd prefer =>.

I will not be doing clean-up like that. I personally see no value in that. Second I copy-pasted most of the code from somewhere else.

---
In reply to: [338819822](https://github.com/dotnet/roslyn/pull/39498#discussion_r338819822) [](ancestors = 338819822)